mlir/lib/Dialect/Rock/utility/transformMapUtils.cpp

```diff
@@ -675,16 +675,67 @@ Value mlir::rock::insertTransposeAndBroadcastTransforms(
     LLVM_DEBUG(llvm::dbgs() << "Reached makeBroadcast with map " << inpIdxMap
                             << " and diff = " << diff << "\n");
 
+    if (diff < 0) {
```

Comment:
I think this is the only required change for the failing case. i.e. more dims in output.

Reply:
Yes, the FoldTranspose changes were a partial fix for a follow-up issue, but I am rethinking the solution. I have reverted FoldTranspose.cpp changes.